homeassistant/components/homekit_controller/device_trigger.py

```diff
@@ -0,0 +1,269 @@
+        self._callbacks.setdefault(iid, []).append(event_handler)
+
+        def async_remove_handler():
+            if iid not in self._callbacks:
+                return
+            self._callbacks[iid].remove(event_handler)
+
+        return async_remove_handler
+
+
+def enumerate_stateless_switch(service):
+    """Enumerate a stateless switch, like a single button."""
+
+    # A stateless switch that has a SERIVCE_LABEL_INDEX is part of a group
+    # And is handled separately
+    if service.has(CharacteristicsTypes.SERVICE_LABEL_INDEX):
+        return []
+
+    char = service[CharacteristicsTypes.INPUT_EVENT]
+
+    # HomeKit itself supports single, double and long presses. But the
+    # manufacturer might not - clamp options to what they say.
+    all_values = clamp_enum_to_char(InputEventValues, char)
+
+    results = []
+    for event_type in all_values:
+        results.append(
+            {
+                "characteristic": char.iid,
+                "value": event_type,
+                "type": "button1",
+                "subtype": HK_TO_HA_INPUT_EVENT_VALUES[event_type],
+            }
+        )
+    return results
+
+
+def enumerate_stateless_switch_group(service):
+    """Enumerate a group of stateless switches, like a remote control."""
+    switches = list(
+        service.accessory.services.filter(
+            service_type=ServicesTypes.STATELESS_PROGRAMMABLE_SWITCH,
+            child_service=service,
+            order_by=[CharacteristicsTypes.SERVICE_LABEL_INDEX],
+        )
+    )
+
+    results = []
+    for idx, switch in enumerate(switches):
+        char = switch[CharacteristicsTypes.INPUT_EVENT]
+
+        # HomeKit itself supports single, double and long presses. But the
+        # manufacturer might not - clamp options to what they say.
+        all_values = clamp_enum_to_char(InputEventValues, char)
+
+        for event_type in all_values:
+            results.append(
+                {
+                    "characteristic": char.iid,
+                    "value": event_type,
+                    "type": f"button{idx + 1}",
+                    "subtype": HK_TO_HA_INPUT_EVENT_VALUES[event_type],
+                }
+            )
+    return results
+
+
+def enumerate_doorbell(service):
+    """Enumerate doorbell buttons."""
+    input_event = service[CharacteristicsTypes.INPUT_EVENT]
+
+    # HomeKit itself supports single, double and long presses. But the
+    # manufacturer might not - clamp options to what they say.
+    all_values = clamp_enum_to_char(InputEventValues, input_event)
+
+    results = []
+    for event_type in all_values:
+        results.append(
+            {
+                "characteristic": input_event.iid,
+                "value": event_type,
+                "type": "doorbell",
+                "subtype": HK_TO_HA_INPUT_EVENT_VALUES[event_type],
+            }
+        )
+    return results
+
+
+TRIGGER_TYPES = {
```

Comment:
This variable has the same name as the one defined at line 18 and could be confusing. Better to rename it.

Reply:
That was a total d'oh! Fixed in latest commits.